android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PwoDiscoveryService.java

```diff
@@ -141,6 +153,38 @@ private void initialize() {
     mHandler = new Handler();
     mUrlToPwoMetadata = new HashMap<>();
     mCanUpdateNotifications = false;
+
+    // Make sure we are trying to load the right version of the cache
```

Comment:
initialize and onDestroy are starting to get a little beefy, so we might want to break them out into other methods at some point.